pandas/io/stata.py

```diff
@@ -1494,7 +1491,11 @@ def read(self, nrows=None, convert_dates=None,
             data = data.set_index(ix)
 
         if columns is not None:
-            data = self._do_select_columns(data, columns)
+            try:
```

Comment:
Another possibility &mdash; though, given your aversion to my wrapping most of `StataReader.read` in a `try/except` block, you might not like it &mdash; is writing a `close_on_error` decorator and wrapping `StataReader.read` &mdash; and maybe `StataWriter.write_file` too &mdash; in it.